Rename MautiCon to Mautic Conference,  update details (#52)
mautic/mautic-community-handbook: pages/08.community-team/02.meetup-guidelines/docs.md

These will be regional in nature and organised with support of the Community Team.

## Mautic Conference

> We are holding annual Mautic Conference events which bring together the international community for a formal conference. 

Reviewer: ```suggestion
We hold annual Mautic Conference events which bring together the international community for a formal conference. 
```

Mautic Conference exists in two flavors:

> * Mautic Conference Global, which is happening online

Reviewer: ```suggestion
* Mautic Conference Global, which is held virtually
```

* Mautic Conference <Continent>, real-life, intended to move to a different continent every year.

> The first such event took place in 2020 - originally planned as an on-premise event, but converted to online due to the 2020 pandemic.

Reviewer: ```suggestion
The first such event took place in 2020 - originally planned as an on-premise event, but converted to a virtual event due to the 2020 pandemic.
```

> Mautic Conference is organised by the community and is inviting sponsors to support the event.

Reviewer: ```suggestion
Mautic Conference is organised by the community and invites sponsors to support the event.
```

> If you are interested in getting involved in any way, please join [#mauticon](https://mautic.slack.com/archives/CA7RCS7SA) on Slack.

Reviewer: ```suggestion
If you are interested in getting involved in any way, please join [#mauticon](https://mautic.slack.com/archives/CA7RCS7SA) on Slack (get an invitation at [https://mautic.org/slack](https://mautic.org/slack)
```
